packages/taquito/src/operations/types.ts

```diff
@@ -568,6 +569,30 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface StakingParams {
+  sourceAndDestination?: string;
+  amount: number;
+  fee?: number;
+  gasLimit?: number;
+  storageLimit?: number;
+  mutez?: boolean;
+}
+
+/**
+ *
+ * @throws {@link InvalidAmountError}
+ */
+export const validateStakingParams = (params: StakingParams, entrypoint: StakingEntrypoint) => {
+  if (params.amount <= 0) {
+    throw new InvalidAmountError(
+      params.amount.toString(),
+      `The 'amount' field in ${entrypoint} should be positive, but ${params.amount} was provided.`
```

Comment:
with previous comments in error class, let's pass `expecting a positive number` . If you insist to be verbose can add after the number

Reply:
it's only validating amount now, abstraction and the name validatingStakingm might reduce readability

Reply:
validation util function can reference how other functions have import `validateAddress` from `@taquito/util`